n_serial.c

```diff
@@ -212,8 +212,9 @@ bool serialNoteReset()
         _Debug("no notecard\n");
 #endif
         _DelayMs(500);
-        _SerialReset();

```

Comment:
A couple years ago Ray changed the serial and I2C reset functions to return a bool:

```
commit 808dd1edca52e28b03fda10af0d59bd3dfccafb8
Author: Ray Ozzie <ray@ozzie.net>
Date:   Wed Aug 12 10:02:21 2020 -0400

    fix

diff --git a/n_lib.h b/n_lib.h
index a8cc84f..ac90f15 100644
--- a/n_lib.h
+++ b/n_lib.h
@@ -40,11 +40,11 @@ bool serialNoteReset(void);
 // Hooks
 void NoteLockNote(void);
 void NoteUnlockNote(void);
-void NoteSerialReset(void);
+bool NoteSerialReset(void);
 void NoteSerialTransmit(uint8_t *, size_t, bool);
 bool NoteSerialAvailable(void);
 char NoteSerialReceive(void);
-void NoteI2CReset(void);
+bool NoteI2CReset(void);
 const char *NoteI2CTransmit(uint16_t DevAddress, uint8_t* pBuffer, uint16_t Size);
 const char *NoteI2CReceive(uint16_t DevAddress, uint8_t* pBuffer, uint16_t Size, uint32_t *avail);
 bool NoteHardReset(void);
```

It seems to me that we wouldn't want to do the retry loop again if we're not able to reset the serial bus. Indeed, that's what's already happening at the start of `serialNoteReset`:

```
if (!_SerialReset()) {
    return false;
}
```

> because it's likely either depended upon or already been worked around

I think it's likely neither of those, actually. As I said, if we've made it this far `_SerialReset` already succeeded once. It seems unlikely to me that it would fail again here, so I would guess no one's actually hitting this in "real life." Also, if it did fail, I would think the bus is in an unusable state, so retrying would have no effect anyway, as transmit/receive aren't going to work.